### ISSUE TITLE
ion-input nested within ion-item doesn't respect --padding-end

### ISSUE THREAD

**Issue by Takkuz** [ionic-team/ionic#17498]
# Bug Report

**Ionic version:**
[x] **4.x**

**Current behavior:**
Creating a floating label ion-input and adding this style
```css
ion-item {
  --padding-start: 0px;
  --padding-end: 0px;
}
```
or this one
```css
ion-item {
    --padding-start: 0px;
    ion-input {
      --padding-end: 0px;
    }
}
```
will results in changing only the "start" side.

**Expected behavior:**
Both padding should follow the padding variable setted

**Steps to reproduce:**
Add a simple IonInput component:
```html
<ion-item>
  <ion-label position="fixed">Floating Label</ion-label>
  <ion-input></ion-input>
</ion-item>
```
with the following CSS:
```css
ion-item {
    --padding-start: 0px;
    --padding-end: 0px;
}
```

**Other information:**
**Ionic info:**
```
Ionic:
   ionic (Ionic CLI)             : 4.10.2
   Ionic Framework               : @ionic/angular 4.0.1
   @angular-devkit/build-angular : 0.12.4
   @angular-devkit/schematics    : 7.2.4
   @angular/cli                  : 7.2.4
   @ionic/angular-toolkit        : 1.3.0

System:
   NodeJS : v10.15.0
   npm    : 6.4.1
   OS     : Linux 4.18
```

**Comment by brandyscarney** [comment]
Thanks for the issue! Item is structured in the following way:

```
<ion-item>
  <slot name="start"></slot>
  <div class="item-inner">
    <div class="input-wrapper">
      <!-- default slot contains label and input -->
      <slot></slot>
    </div>
    <slot name="end"></slot>
  </div>
</ion-item>
```

So when you are styling `--padding-start` and `--padding-end` you are changing the padding on the host element `ion-item`. I believe the padding you are seeing is coming from the default padding on the `.item-inner` which allows us to pad the items in the `end` slot without losing the bottom border that stretches all of the way to the right.

What you're looking for is the following:

```css
ion-item {
  --padding-start: 0;

  --inner-padding-end: 0;
}
```

See the CSS Custom properties for item here: https://ionicframework.com/docs/api/item#css-custom-properties

**Comment by Takkuz** [comment]
Thank you and sorry for the useless issue, I already found it lookin at the inspector but this should be better documented because I was looking at the `ion-input` page where there is no reference to the `ion-item`. It wasn't neither logical to think that, looking at a document page I need to look in another page to fully access all the options of those component without reference.

**Comment by brandyscarney** [comment]
@Takkuz Thanks for the suggestion! Please open an issue on our [ionic-docs](https://github.com/ionic-team/ionic-docs/issues) repository with any ideas for how we can improve the documentation. 🙂

**Comment by KGMaxey** [comment]
@brandyscarney 
Hi! Sorry to jump on the end of this conversation. I've been messing with ionic for a few days and I'm still learning styling using a shadow dom so please forgive me if I'm missing something obvious. After digging into the repository I've found where the original styling for the ion-item is set (core/src/components/item/). As shown below, ionic adds a start padding and then an inner end padding, however it doesn't have complementing end padding and inner start padding to default centering items.

**core/src/components/item/item.ios.scss**
```
 --padding-end: #{$item-ios-padding-end};
 --inner-padding-start: #{$item-ios-padding-start / 2};
```

I wish I had seen this thread before I dug so far into this as your suggestion below would have solved my problem.

**variable.scss**
```
ion-item {
  --padding-start: 0;
  --inner-padding-end: 0;
}
```

Just to test, I followed the same pattern and imported the css variables into my variables.scss and used this to see what happens and it centers and is padded perfectly. I'm just curious as to why this isn't the default behavior for ion-items? (Or vice-versa to force user to set <ion-item padding>)

**Default Padded Solution In variable.scss**
```
ion-item {
    --padding-start: #{$item-ios-padding-start};
    --inner-padding-end: #{$item-ios-padding-end / 2};
    --padding-end: #{$item-ios-padding-end};
    --inner-padding-start: #{$item-ios-padding-start / 2};
}
```

Thank you very much for your time and great work to the whole team, so far Ionic is awesome!

**Comment by brandyscarney** [comment]
Hi @KGMaxey! No worries. I understand it's a bit confusing without seeing the structure or inspecting the element, which is complex because of shadow DOM. We really do want to make this better documented and as easy to use as possible.

So I am not sure that I 100% understand your question so please correct me if I'm misunderstanding, but using your CSS gives me the following item:

<img width="389" alt="screen shot 2019-02-27 at 10 11 22 am" src="https://user-images.githubusercontent.com/6577830/53500305-12f45600-3a78-11e9-8a5b-89d0d3b42615.png">

Notice how the line on the right doesn't extend to the edge of the screen? The reason is because the border for an item with `inset` lines is actually applied on the inner item div and not the host `ion-item`. This is a problem because we need to be able to pad the border on the left to account for any elements in the `start` slot and *not* be under those, while remaining extended to the edge on the right. 

Basically the problem is that in order to provide the `inset` lines and match the spec, we need to be able to have inner padding on the left, but we don't want the content to be on the right edge of the screen so we add some padding on the host element. Here's an example of the MD spec so you can better see what I mean (notice the divider lines):

<img src="https://user-images.githubusercontent.com/6577830/53500635-bfced300-3a78-11e9-96d5-c8481710b67d.png" width="400px">

We try to follow the iOS and MD spec as much as possible out of the box, but still allow for customizations if desired.

Please let me know if you have questions, thank you!
